zaraz.go

```diff
@@ -0,0 +1,234 @@
+	if err != nil {
+		return ZarazConfigResponse{}, err
+	}
+
+	var recordResp ZarazConfigResponse
+	err = json.Unmarshal(res, &recordResp)
+	if err != nil {
+		return ZarazConfigResponse{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return recordResp, nil
+}
+
+func (api *API) UpdateZarazConfig(ctx context.Context, rc *ResourceContainer, params UpdateZarazConfigParams) (ZarazConfigResponse, error) {
+	if rc.Identifier == "" {
+		return ZarazConfigResponse{}, ErrMissingZoneID
+	}
+
+	uri := fmt.Sprintf("/zones/%s/settings/zaraz/v2/config", rc.Identifier)
+	res, err := api.makeRequestContext(ctx, http.MethodPut, uri, params)
+	if err != nil {
+		return ZarazConfigResponse{}, err
+	}
+
+	var updateResp ZarazConfigResponse
+	err = json.Unmarshal(res, &updateResp)
+	if err != nil {
+		return ZarazConfigResponse{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return updateResp, nil
+}
+
+func (api *API) GetZarazWorkflow(ctx context.Context, rc *ResourceContainer) (ZarazWorkflowResponse, error) {
+	if rc.Identifier == "" {
+		return ZarazWorkflowResponse{}, ErrMissingZoneID
+	}
+
+	uri := fmt.Sprintf("/zones/%s/settings/zaraz/v2/workflow", rc.Identifier)
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return ZarazWorkflowResponse{}, err
+	}
+
+	var response ZarazWorkflowResponse
+	err = json.Unmarshal(res, &response)
+	if err != nil {
+		return ZarazWorkflowResponse{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return response, nil
+}
+
+func (api *API) UpdateZarazWorkflow(ctx context.Context, rc *ResourceContainer, workflowToUpdate string) (ZarazWorkflowResponse, error) {
+	if rc.Identifier == "" {
+		return ZarazWorkflowResponse{}, ErrMissingZoneID
+	}
+
+	uri := fmt.Sprintf("/zones/%s/settings/zaraz/v2/workflow", rc.Identifier)
+	res, err := api.makeRequestContext(ctx, http.MethodPut, uri, workflowToUpdate)
+	if err != nil {
+		return ZarazWorkflowResponse{}, err
+	}
+
+	var response ZarazWorkflowResponse
+	err = json.Unmarshal(res, &response)
+	if err != nil {
+		return ZarazWorkflowResponse{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return response, nil
+}
+
+func (api *API) PublishZarazConfig(ctx context.Context, rc *ResourceContainer, description string) (ZarazPublishResponse, error) {
+	if rc.Identifier == "" {
+		return ZarazPublishResponse{}, ErrMissingZoneID
+	}
+
+	uri := fmt.Sprintf("/zones/%s/settings/zaraz/v2/publish", rc.Identifier)
+	res, err := api.makeRequestContext(ctx, http.MethodPost, uri, description)
+	if err != nil {
+		return ZarazPublishResponse{}, err
+	}
+
+	var response ZarazPublishResponse
+	err = json.Unmarshal(res, &response)
+	if err != nil {
+		return ZarazPublishResponse{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return response, nil
+}
+
+func (api *API) GetZarazConfigHistory(ctx context.Context, rc *ResourceContainer, params GetZarazConfigHistoryParams) ([]ZarazConfigRow, *ResultInfo, error) {
```

Comment:
generally a "fetch all" type of method is prefixed with `List` to denote it is capable of returning multiple values - https://github.com/cloudflare/cloudflare-go/blob/master/docs/experimental.md#consistent-crud-method-signatures